airflow/executors/__init__.py

```diff
@@ -57,32 +61,21 @@ def get_default_executor():
     return DEFAULT_EXECUTOR
 
 
-class Executors:
-    LocalExecutor = "LocalExecutor"
-    SequentialExecutor = "SequentialExecutor"
-    CeleryExecutor = "CeleryExecutor"
-    DaskExecutor = "DaskExecutor"
-    KubernetesExecutor = "KubernetesExecutor"
-
-
 def _get_executor(executor_name):
     """
     Creates a new instance of the named executor.
     In case the executor name is not know in airflow,
     look for it in the plugins
     """
-    if executor_name == Executors.LocalExecutor:
+    if executor_name == AvailableExecutors.LOCAL_EXECUTOR:
         return LocalExecutor()
-    elif executor_name == Executors.SequentialExecutor:
+    elif executor_name == AvailableExecutors.SEQUENTIAL_EXECUTOR:
         return SequentialExecutor()
-    elif executor_name == Executors.CeleryExecutor:
-        from airflow.executors.celery_executor import CeleryExecutor
+    elif executor_name == AvailableExecutors.CELERY_EXECUTOR:
```

Comment:
It is very important that the modules are imported here. Otherwise, all possible executors will load on every person, which causes a drastic increase in the time it takes to run the application.  This means that everyone must also have all executors installed.

In this PR: https://github.com/apache/airflow/pull/6594  In this PR, I have reduced the number of modules loaded and you can see how it affected performance.

Reply:
Yes. It caused mutiple other problems. But it was wrong originally that only some of the executors were locally imported (and that was part of the problems with cyclic imports). Now all of them are locally imported. I don't think it has an impact on performance. The classes are too small to make a difference. I think optimising for performance here would be a premature optimisation. However cyclic imports should be avoided and I hope I can make all the tests succeed now.

Reply:
This has a big impact on performance because one class can load more classes, and each class must be initialized. Airflow separate tasks are run in separate processes, which means that classes are loaded repeatedly. If possible, we should limit the classes that can be loaded.